src/PrestaShopBundle/Controller/Admin/Sell/Catalog/MonitoringController.php

```diff
@@ -0,0 +1,250 @@
+        ProductWithoutImageFilters $productWithoutImageFilters,
+        ProductWithoutDescriptionFilters $productWithoutDescriptionFilters,
+        ProductWithoutPriceFilters $productWithoutPriceFilters
+    ) {
+        $deleteCategoryForm = $this->createForm(DeleteCategoriesType::class);
+
+        $emptyCategoryGrid = $this->getEmptyCategoryGrid($emptyCategoryFilters);
+        $productWithCombinationGrid = $this->getProductWithCombinationGrid($productWithCombinationFilters);
+        $productWithoutCombinationGrid = $this->getProductWithoutCombinationGrid($productWithoutCombinationFilters);
+        $disabledProductGrid = $this->getDisabledProductGrid($disabledProductFilters);
+        $productWithoutImageGrid = $this->getProductWithoutImageGrid($productWithoutImageFilters);
+        $productWithoutDescriptionGrid = $this->getProductWithoutDescriptionGrid($productWithoutDescriptionFilters);
+        $productWithoutPriceGrid = $this->getProductWithoutPriceGrid($productWithoutPriceFilters);
+
+        return $this->render('@PrestaShop/Admin/Sell/Catalog/Monitoring/index.html.twig', [
+            'enableSidebar' => true,
+            'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+            'emptyCategoryGrid' => $this->presentGrid($emptyCategoryGrid),
+            'deleteCategoryForm' => $deleteCategoryForm->createView(),
+            'productWithCombinationGrid' => $this->presentGrid($productWithCombinationGrid),
+            'productWithoutCombinationGrid' => $this->presentGrid($productWithoutCombinationGrid),
+            'disabledProductGrid' => $this->presentGrid($disabledProductGrid),
+            'productWithoutImageGrid' => $this->presentGrid($productWithoutImageGrid),
+            'productWithoutDescriptionGrid' => $this->presentGrid($productWithoutDescriptionGrid),
+            'productWithoutPriceGrid' => $this->presentGrid($productWithoutPriceGrid),
+        ]);
+    }
+
+    /**
+     * Provides filters functionality
+     *
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))")
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function searchAction(Request $request)
+    {
+        $gridIdentifiers = $this->identifySearchableGrid($request);
+
+        /** @var ResponseBuilder $responseBuilder */
+        $responseBuilder = $this->get('prestashop.bundle.grid.response_builder');
+
+        return $responseBuilder->buildSearchResponse(
+            $gridIdentifiers['grid_definition'],
+            $request,
+            $gridIdentifiers['grid_id'],
+            'admin_monitoring_index'
+        );
+    }
+
+    /**
+     * Parses grid identifying parts from request in order to recognize which grid is being filtered
+     *
+     * @param Request $request
+     *
+     * @return array
+     */
+    private function identifySearchableGrid(Request $request)
```

Comment:
@matks, I think the behavior of multi-grid per page filtering is not yet considered when talking about extensions?
For example, some module introduces new grid to page, i guess it will ruin filtering of that page grids unless we ensure to always check which grid is being filtered (including the ones from modules if they exists) ?

Reply:
You're thinking of a module that *adds* a filterable grid to a page ?
This is an heavy case, so I think the module developer can expect there that he'll need to create a dedicated Controller and View.

That is an interesting question nonetheless. Can you open an issue about it ?